x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -98,23 +144,80 @@ export const DetailsPageMappingsContent: FunctionComponent<{
 
   useMappingsStateListener({ value: parsedDefaultValue, status: 'disabled' });
 
-  const {
-    fields: { byId, rootLevelFields },
-    search,
-    documentFields,
-  } = useMappingsState();
+  const onCancelAddingNewFields = useCallback(() => {
+    setAddingFields(!isAddingFields);
+
+    //  reset mappings to previous state
+    dispatch({
+      type: 'editor.replaceMappings',
+      value: {
+        ...previousState,
+        documentFields: {
+          status: 'disabled',
+          editor: 'default',
+        },
+      },
+    });
+  }, [isAddingFields, dispatch, previousState]);
+
+  const addFieldButtonOnClick = useCallback(() => {
+    setAddingFields(!isAddingFields);
+
+    // when adding new field, save previous state. This state is then used by FieldsList component to show only saved mappings.
+    setPreviousStateFields(getFieldsFromState(state));
+    setPreviousState(state);
+
+    // reset mappings and change status to create field.
+    dispatch({
+      type: 'editor.replaceMappings',
+      value: {
+        ...state,
+        fields: { ...state.fields, byId: {}, rootLevelFields: [] } as NormalizedFields,
+        documentFields: {
+          status: 'creatingField',
+          editor: 'default',
+        },
+      },
+    });
+  }, [dispatch, isAddingFields, state]);
+
+  const updateMappings = useCallback(async () => {
+    try {
+      const { error } = await updateIndexMappings(indexName, deNormalize(state.fields));
+
+      if (!error) {
+        notificationService.showSuccessToast(
+          i18n.translate('xpack.idxMgmt.indexDetails.mappings.successfullyUpdatedIndexMappings', {
+            defaultMessage: 'Index Mapping was successfully updated',
+          })
+        );
+        refetchMapping();
+      } else {
+        setSaveMappingError(error.message);
```

Comment:
We could add some logic to reset the error, when the user cancel the form or re-sends the request, I think currently the error will just stay on the page until re-load
<img width="855" alt="Screenshot 2024-03-26 at 14 19 45" src="https://github.com/elastic/kibana/assets/6585477/fc5e7c4a-db69-47d7-b6cc-885b0831b270">

Reply:
Currently this error will be cleared when update mappings api is success and new mappings are fetched. 

I think this behaviour is similar to index/component template mappings flow. That is when there is an error, it would be cleared only templates are saved. 

https://github.com/elastic/kibana/assets/55930906/be571b6b-ee57-4364-8edc-be5ac0d5ebd3

IMO the ideal behaviour would be clear this error when the error is corrected in edit form flyout.

Reply:
Ticket to track this work : https://github.com/elastic/enterprise-search-team/issues/7224